library_information_validator.py

```diff
@@ -0,0 +1,318 @@
+                )
+            if(self.prep_standards(info)[6] != None):
+                logger.info(
+                'Sample QC recommendation: submission of {} data'.format(self.prep_standards(info)[6])
+                )
+            logger.info(
+                'Checked entry in sample concentration and volume. {}/{} pass, {} warning(s).'\
+                .format(passes, total, WARNINGS)
+                )
+
+
+
+
+class Validator(object):
+# Initializer / Instance attributes
+    def __init__(self, access_sample_info_sheet, concentrationID, volumeID, rinID):
+        self.access_sample_info_sheet = access_sample_info_sheet
+        self.concentrationID = concentrationID
+        self.volumeID = volumeID
+        self.rinID = rinID
+
+# instance methods
+    def validate_numeric(self):
+        """Checks whether value is numeric or not."""
+#        logging.debug(self.access_sample_info_sheet[self.concentrationID].value, type(self.access_sample_info_sheet[self.concentrationID].value))
+    #    if type(self.concentration)!= 'n':
+        for checkNumbers in [self.concentrationID, self.volumeID, self.rinID]:
+            if not isinstance(self.access_sample_info_sheet[checkNumbers].value, numbers.Number):
+                try:
+                    float(self.access_sample_info_sheet[checkNumbers].value.replace(",", "."))
+                    logger.error(
+                        'Cell {} with value \"{}\" is not numeric due to decimal point/comma clash.'\
+                        .format(self.access_sample_info_sheet[checkNumbers].coordinate, self.access_sample_info_sheet[checkNumbers].value)
+                        )
+                    return False
+                except ValueError:
+                    logger.error(
+                    'Cell {} with value \"'.format(self.access_sample_info_sheet[checkNumbers].coordinate)+ self.access_sample_info_sheet[checkNumbers].value + '\" is not numeric'
+                    )
+                except TypeError:
+                    if self.access_sample_info_sheet[checkNumbers].value is None:
+                        logger.error(
+                        'Cell {} is numeric but empty'.format(self.access_sample_info_sheet[checkNumbers].coordinate)
+                        )
+                        return False
+                    else:
+                        raise
+            return False
+
+    def validate_conc(self, min_conc, max_conc):
+        if(self.access_sample_info_sheet[self.concentrationID].value < min_conc) \
+        or (self.access_sample_info_sheet[self.concentrationID].value > max_conc):
+            global WARNINGS
+            WARNINGS = WARNINGS+1
+
+            logger.warning('Sample concentration ({}ng/ul) in cell {} is out of specifications: {}-{}ng/ul'\
+            .format(self.access_sample_info_sheet[self.concentrationID].value,self.concentrationID, min_conc, max_conc))
+        return True
+
+    def validate_vol(self, vol):
```

Comment:
Sometimes, vol is actually a range. I don't think we handle that properly, and I don't know how we should do that.